### PR TITLE
Updates section navigation

### PR DESCRIPTION
**Pull request by h-enk**
## Summary

Brief explanation of the proposed changes.

- [Sidebar](https://getdoks.org/docs/basics/navigation/#sidebar)
- Auto generate section menu (default), or
- Configure section menu (partly automated):

```toml
[[sidebar_docs]]
  name = "Start here"
  pageRef = "/docs/start-here"
  weight = 10

[[sidebar_docs]]
  name = "Guides"
  pageRef = "/docs/guides"
  weight = 20

[[sidebar_docs]]
  name = "Reference"
  pageRef = "/docs/reference"
  weight = 30
```

Sections are not collapsed (default), override per section in the frontmatter of the section's `_index.md`:

```md
---
sidebar:
  collapsed: true
---
```

Works with:

```bash
pnpm run create --kind docs guides
```

## Basic example

Include a basic example, screenshots, or links.

Snag_1481292

## Motivation

Why are we doing this? What use cases does it support? What is the expected outcome?

More robust, simplified, and powerful code with more features and easier expandability possibilities for for example badges, icons, and `target` attribute.

## Checks

- [ ] Read [Create a Pull Request](https://gethyas.com/docs/contributing/how-to-contribute/#create-a-pull-request)
- [ ] Supports all screen sizes (if relevant)
- [ ] Supports both light and dark mode (if relevant)
- [ ] Passes `npm run test`

### CODE REVIEW

File: layouts/_default/single.html [hunk 1]
```diff
@@ -2,9 +2,7 @@
 	<div class="row flex-xl-nowrap">
 		{{ if (in site.Params.doks.sectionNav .Section) -}}
 		<div class="col-lg-5 col-xl-4 docs-sidebar{{ if ne site.Params.doks.navbarSticky true }} docs-sidebar-top{{ end }}{{ if site.Params.doks.headerBar }} docs-sidebar-offset{{ end }} d-none d-lg-block">
-			<nav {{ if eq site.Params.doks.menu.section.collapsibleSidebar false }}id="sidebar-default" {{ end }}class="docs-links" aria-label="Main navigation">
-				{{ partial "sidebar/docs-menu.html" . }}
-			</nav>
+			{{ partial "sidebar/section-menu.html" . }}
```
Comment (kosssi) [inline layouts/_default/single.html:5]: Maybe I don't have understand but this parameter is not used `site.Params.doks.menu.section.collapsibleSidebar`?

Comment (h-enk) [reply layouts/_default/single.html:5]: Correct. Now, when no `sidebar_<section>` menu entries are found, Doks automatically generates a sidebar based on the filesystem structure of your documentation — no parameter needed. The new setup is a lot smarter and more robust.

Comment (kosssi) [reply layouts/_default/single.html:5]: Thanks for your response, I don't understand how can I get the same design like this before this commit :

image

not like this (last version)

image

I don't want colapse menu ;)

**Comment by kosssi** [comment]
I don't want to deviate too much from Doks otherwise it will complicate the update.

I especially wanted you to be aware if you hadn't seen the problem ;)

**Comment by h-enk** [comment]
I understand that — thanks

**Pull request merged**